devito/passes/clusters/aliases.py

```diff
@@ -211,9 +216,19 @@ def extract(cls, n, context, min_cost, max_alias, cluster, sregistry):
     def in_writeto(cls, max_par, dim, cluster):
         return PARALLEL in cluster.properties[dim]
 
+
+class CallbacksDivs(CallbacksInvariants):
+
+    mode = 'divs'
+
+    @classmethod
+    def _extract_rule(cls, context, min_cost, cluster):
+        return lambda e: (e.is_Pow and e.exp.is_Integer and e.exp < 0 and
+                          all(i.function.is_const for i in e.base.free_symbols))
```

Comment:
So functions divs not considered?

Reply:
they will be picked already by `cire(..., 'invariants', ...)`

rationale: here I'm looking for scalar divisions. These are negligible when running on CPUs (the jit compiler will hoist them out of the innermost loop, so their cost will be completely amortized out over a bunch of loop iterations), but they typically have an impact on GPUs, since it would be one division per GPU thread.